sdk/storage/Azure.Storage.DataMovement.Files.Shares/src/ShareFileStorageResource.cs

```diff
@@ -199,13 +203,60 @@ protected override async Task<StorageResourceItemProperties> GetPropertiesAsync(
             Response<ShareFileProperties> response = await ShareFileClient.GetPropertiesAsync(
                 conditions: _options?.SourceConditions,
                 cancellationToken: cancellationToken).ConfigureAwait(false);
-            if (ResourceProperties == default)
+            if (ResourceProperties != default)
+            {
+                ResourceProperties.AddToStorageResourceItemProperties(response.Value);
+            }
+            else
             {
                 ResourceProperties = response.Value.ToStorageResourceItemProperties();
             }
             return ResourceProperties;
         }
 
+        protected override async Task<string> GetPermissionsAsync(
+            StorageResourceItemProperties properties = default,
+            CancellationToken cancellationToken = default)
+        {
+            string permissionKey = properties?.RawProperties?.GetSourcePermissionKey();
+            if (!string.IsNullOrEmpty(permissionKey))
+            {
+                ShareClient parentShare = ShareFileClient.GetParentShareClient();
+                return await parentShare.GetPermissionAsync(permissionKey, cancellationToken).ConfigureAwait(false);
+            }
+            return default;
+        }
+
+        protected override async Task SetPermissionsAsync(
+            StorageResourceItem sourceResource,
+            StorageResourceItemProperties sourceProperties,
+            CancellationToken cancellationToken = default)
+        {
+            if (sourceResource is ShareFileStorageResource)
+            {
+                if (_options?.FilePermissions?.Preserve ?? false)
+                {
+                    ShareFileStorageResource sourceShareFile = (ShareFileStorageResource)sourceResource;
+                    string destinationPermissionKey = sourceProperties?.RawProperties?.GetDestinationPermissionKey();
+                    if (destinationPermissionKey == default)
+                    {
+                        string sourcePermissions = await sourceShareFile.GetPermissionsAsync(sourceProperties, cancellationToken).ConfigureAwait(false);
+
+                        if (!string.IsNullOrEmpty(sourcePermissions))
+                        {
+                            ShareClient parentShare = ShareFileClient.GetParentShareClient();
+                            PermissionInfo permissionsInfo = await parentShare.CreatePermissionAsync(sourcePermissions, cancellationToken).ConfigureAwait(false);
+                            _destinationPermissionKey = permissionsInfo.FilePermissionKey;
```

Comment:
Should we instead write this to the `properties.RawProperties` so we don't have to save it separately? Would also simplify the call to the `GetFileSmbProperties` extension if it was passed in the properties. The `properties` _should_ always be set by now because we call `GetProperties` in the job before this, right?

Eventually it would need to go into that properties bag so it can be checkpointed anyway.